packages/transformers/babel/src/babel7.js

```diff
@@ -1,37 +1,49 @@
 // @flow
 
 import type {LocalRequire, MutableAsset, AST} from '@parcel/types';
+import packageJson from '../package.json';
+import invariant from 'assert';
+
+const transformerVersion: mixed = packageJson.version;
+invariant(typeof transformerVersion === 'string');
 
 export default async function babel7(
   asset: MutableAsset,
   localRequire: LocalRequire,
   options: any
 ): Promise<?AST> {
-  let config = options.config;
-
   // If this is an internally generated config, use our internal @babel/core,
   // otherwise require a local version from the package we're compiling.
   let babel = options.internal
     ? require('@babel/core')
     : await localRequire('@babel/core', asset.filePath);
 
-  config.code = false;
-  config.ast = true;
-  config.filename = asset.filePath;
-  config.babelrc = false;
-  config.configFile = false;
-  config.parserOpts = Object.assign({}, config.parserOpts, {
-    allowReturnOutsideFunction: true,
-    strictMode: false,
-    sourceType: 'module',
-    plugins: ['dynamicImport']
-  });
+  let config = {
+    ...options.config,
+    code: false,
+    ast: true,
+    filename: asset.filePath,
+    babelrc: false,
+    configFile: false,
+    parserOpts: {
+      ...options.config.parserOpts,
+      allowReturnOutsideFunction: true,
+      strictMode: false,
+      sourceType: 'module',
+      plugins: ['dynamicImport']
+    },
+    caller: {
+      name: 'parcel',
+      version: transformerVersion,
+      targets: JSON.stringify(options.targets)
+    }
+  };
 
   let code = await asset.getCode();
 
   let res;
   if (asset.ast) {
-    res = babel.transformFromAst(asset.ast.program, code, config);
+    res = babel.transformFromAstSync(asset.ast.program, code, config);
```

Comment:
This behaves async if a callback is given, and may be removed at some point.

https://babeljs.io/docs/en/babel-core#transformfromast